src/main/java/io/strimzi/kafka/quotas/ThrottleFactorPolicy.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package io.strimzi.kafka.quotas;
+
+/**
+ * Determines the current restriction factor to be applied to the client quota.
+ * Values returned are required to be in the range <code>0.0..1.0</code> inclusive.
+ * Where a value of `1.0` implies no additional restriction over and above the defined quota.
+ * A value of `0.0` implies that there is no quota available regardless of the defined quota.
+ */
+public interface ThrottleFactorPolicy extends VolumeObserver {
+    /**
+     * Register a listener to be invoked when the throttle factor is changed.
+     * @param listener to be executed whenever the factor is updated.
+     */
+    void addUpdateListener(Runnable listener);
+
+    /**
+     * @return The current factor in the range {@code [0..1]} to scale the throttle by.
+     */
+    Double currentFactor();
```

Comment:
None of the existing policies seem to require null, and the meaning of null is not defined, so why is this not a primitive `double`?

Reply:
resolved with 434502c53e82bd61cbc8ad3a82a7cd9a0dbaf431